vagrant/ansible/roles/samba-glusterfs.setup/tasks/main.yml

```diff
@@ -64,6 +64,12 @@
         state: absent
   run_once: true
 
+- name: Add acl_xattr VFS module
+  lineinfile:
+    dest: /etc/samba/smb.conf
+    line: "vfs objects = acl_xattr glusterfs"
```

Comment:
Are we not adding this to the template because setting `vfs objects` is done by the gluster-ansible modules? Or is it done by the bloody hook scripts? ;-)

Reply:
I believe the hook scripts could also be causing issues with the multiple run failures.  At least one failure in my system has the ctdb stop script attempt to unmount the lock filesystem and then remove entry from fstab irrespective of the status of the unmount command. The start script then mounts the lock volume and following the success of the mount, adds the entry into fstab.

The problem is that when ctdb is running, it has a lock on the recovery lock on the lock volume. The stop scripts are unable to unmount the volume but remove the fstab entry. The start scripts then attempt to mount the volume, but fail because it is already mounted. So it doesn't create the fstab entry. 

So the master node now doesn't have the fstab entry to mount the recovery lock volume.

Reply:
Yeah, the hooks scripts are among the most horrid and dangerous inventions. I was always entirely against them. I assume that they are enabled by gluster-ansible. I prefer we try to prevent that / disable them, and take full control...

Reply:
> Yeah, the hooks scripts are among the most horrid and dangerous inventions. I was always entirely against them. I assume that they are enabled by gluster-ansible.

There are mainly two types of hook scripts that we are dealing with Samba-CTDB use case.
- set of scripts which add/remove/modify GlusterFS volume share section within _smb.conf_. This involvement can be disabled by **not** setting `user.smb` volume option.
- set of scripts which add/remove fstab entry for CTDB lock volume and correspondingly mount and unmount it. This is enabled once we modify a particular variable in those scripts to point to CTDB lock volume name. _glutser-ansible_ currently performs this variable assignment within _gluster-ansible-features_.

> I prefer we try to prevent that / disable them, and take full control...

Fine by me.